src/components/App/App.js

```diff
@@ -1,19 +1,66 @@
-import ErrorBoundary from '../ErrorBoundary/ErrorBoundary';
-import Footer from '../Footer/Footer';
-import Header from '../Header/Header';
-import Main from '../Main/Main';
-import './App.css';
+import { useState, useEffect } from "react";
+import "./App.css";
+
+import ErrorBoundary from "../ErrorBoundary/ErrorBoundary";
+import Footer from "../Footer/Footer";
+import Header from "../Header/Header";
+import Main from "../Main/Main";
+import MovieDetails from "../MovieDetails/MovieDetails";
+
+import { AppContext } from "../../hooks/AppContext";
+import { useDummyHttp } from "../../hooks/DummyHttp";
 
 const App = () => {
-	return (
-		<ErrorBoundary>
-			<div className="app">
-				<Header />
-				<Main />
-				<Footer />
-			</div>
-		</ErrorBoundary>
-	);
+  const [isHeaderVisible, setIsHeaderVisible] = useState(true);
+  const [movie, setMovie] = useState({});
+  const [isLoading, movies, setMovies, moviesNumber, setMoviesNumber] =
+    useDummyHttp();
+
+  const appContext = {
+    moviesContext: {
+      movies,
+      setMovies,
+      moviesNumber,
+      setMoviesNumber,
+    },
+    movieContext: {
+      movie,
+      setMovie,
+    },
+    navigationContext: {
+      isHeaderVisible,
+      setIsHeaderVisible,
+    },
+  };
+
+  let fetchedMovieContent = (
+    <div className="loading-message__container ">
+      <h2 className="loading-message">Loading...</h2>
+    </div>
+  );
+
+  if (!isLoading) {
+    fetchedMovieContent = (
+      <>
```

Comment:
1. Just <Main/>
2. You can move loading indication inside Main

Reply:
done